File: src/Application Specific/Classes/Entity Collections/Site/SiteCustomerAudit/SiteCustomerAuditValidator.vb
Namespace Entity

Namespace SiteCustomerAudits

Public Class SiteCustomerAuditValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oSiteCustomerAudit As SiteCustomerAudit)

        'make sure that contact object is valid
        If oSiteCustomerAudit.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oSiteCustomerAudit.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Site/SiteCustomerAudit/SiteCustomerAuditValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace SiteCustomerAudits
    {
        // make sure that contact object is valid
        public class SiteCustomerAuditValidator : BaseValidator
        {
            public void Validate(SiteCustomerAudit oSiteCustomerAudit)
            {
                // make sure that contact object is valid
                if (oSiteCustomerAudit.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oSiteCustomerAudit.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}